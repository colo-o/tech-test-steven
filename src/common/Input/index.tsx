import classNames from 'classnames';
import { ChangeEvent, forwardRef } from 'react';

type InputProps = {
  placeholder?: string;
  name: string;
  value?: string;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  type: 'text' | 'number' | 'password';
  className?: string;
};

type InputRef = HTMLInputElement | null;

const Input = forwardRef<InputRef, InputProps>(
  ({ value, onChange, name, placeholder, type, className }, ref) => {
    const inputClassName = classNames(
      `${className} bg-darkGray border-none rounded-full px-8 py-4 text-white caret-primary w-full shadow-lg shadow-bgDark focus:outline-none`,
      {}
    );

    return (
      <input
        ref={ref}
        value={value}
        type={type}
        placeholder={placeholder}
        onChange={(event: ChangeEvent<HTMLInputElement>) => onChange(event)}
        name={name}
        aria-label={name}
        className={inputClassName}
      />
    );
  }
);

Input.displayName = 'Input Text';

export default Input;
