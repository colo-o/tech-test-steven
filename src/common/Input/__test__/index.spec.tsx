import { render, screen, fireEvent } from '@testing-library/react';

import Input from 'common/Input';
import { useState } from 'react';

const Wrapp = () => {
  const [value, setValue] = useState('');

  return (
    <Input
      onChange={ev => setValue(ev.target.value)}
      value={value}
      name="input"
      type="text"
    />
  );
};

describe('[INPUT]', () => {
  it('Should render a input value when use types something', () => {
    render(<Wrapp />);

    const input: HTMLInputElement = screen.getByLabelText('input');
    fireEvent.change(input, { target: { value: 'Hello world' } });

    expect(input.value).toBe('Hello world');
    expect(input).toMatchSnapshot();
  });
});
