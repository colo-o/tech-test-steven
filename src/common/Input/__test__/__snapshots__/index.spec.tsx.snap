// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[INPUT] Should render a input value when use types something 1`] = `
<input
  aria-label="input"
  class="undefined bg-darkGray border-none rounded-full px-8 py-4 text-white caret-primary w-full shadow-lg shadow-bgDark focus:outline-none"
  name="input"
  type="text"
  value="Hello world"
/>
`;
